pkg/api/http.go

```diff
@@ -33,27 +35,6 @@ func versionMiddleware(next http.Handler) http.Handler {
 	})
 }
 
-// TODO: use Istio tracing package
-// https://github.com/istio/istio/blob/master/pkg/tracing/config.go
-func copyTracingHeaders(from *http.Request, to *http.Request) {
-	headers := []string{
-		"x-request-id",
```

Comment:
Can you please add a test for this header propagation?

Reply:
Let me try. It would be awkward as the tracer used for mock server is a noop tracer. The child span does turn up inside the parent span so it does work. Hard to prove though.

Reply:
Have added the code back as on debugging I see some Opentelemetry headers passed but not the specific headers being copied.